### ISSUE TITLE
session closed: 0, reason: The decoded text message was too big for the output buffer and the endpoint does not support partial messages

### ISSUE THREAD

**Issue by rahulsom** [seratch/jslack#92]
Long-time user, first-time reporter here. Thanks for building this library. It is great.

From time to time, my app is rendered useless with this message

```
2019-02-06 14:30:49.982 DEBUG [ient-SecureIO-1]               c.g.s.j.a.r.RTMClient                   -- session closed: 0, reason: The decoded text message was too big for the output buffer and the endpoint does not support partial messages
```

Once that happens, the `com.github.seratch.jslack.Slack` instance needs to be discarded. Since I'm creating it as a Spring Bean, and the instance is autowired, I end up restarting the application.

My current close handler only updates the health of the application and lets the unhealthy check take care of restarting the application.

```kotlin
            rtm.addCloseHandler {
                log.error("Session Closed: $it")
                slackHealth.healthy = false
                slackHealth.reason = it.reasonPhrase
            }
```

1. Is there a strategy to recover from this?
2. Can this problem be avoided?

**Comment by seratch** [comment]
I've not investigated your situation on my end yet but, at least, I can say that you don't need to reboot the entire application. `RTMClient` doesn't depend on any states inside your singleton Slack instance except for `wssUrl`. So, I guess, when you have the unexpected shut down of the RTM client, you should be able to recover it by just calling `RTMClient#disconnect()` then `RTMClient#connect()`. The `connect` method establishes a new session with the server. https://github.com/seratch/jslack/blob/1.1.7/src/main/java/com/github/seratch/jslack/api/rtm/RTMClient.java#L31-L35

**Comment by rahulsom** [comment]
Weirdly, that did not work. 
I stopped making `Slack` a bean and instead created a holder that was a bean, had it discard and recreate the `Slack` object and initialize it.
That works, and for now, my problem is gone.

I'll leave this open to investigate the `The decoded text message was too big for the output buffer and the endpoint does not support partial messages` error.

Thanks!

**Comment by seratch** [comment]
Thank you for your response. I'm fine to leave this issue open, too.

**Comment by seratch** [comment]
I realized that calling `RTMClient#disconnect()` then `RTMClient#connect()`surely doesn;t work for the case. So, since version 1.3.0, I've added [`RTMClient#reconnect()`](https://github.com/seratch/jslack/blob/1.3.0/src/main/java/com/github/seratch/jslack/api/rtm/RTMClient.java#L101-L117) which works for any cases.

Also, I applied the changes to `RTMClient` regarding its behavior when receiving some errors. Previously, the client used to abandon the underlying WebSocket session when it gets errors, anyway. But I found that it's not a proper way to handle errors. Therefore, since version 1.3.0, the default error handler no longer does the removal of sessions. The change may be beneficial for your use cases.
